### PR TITLE
PART command

### PR DESCRIPTION
**Pull request by theovgl**

### CODE REVIEW

File: src/Message.cpp [hunk 1]
```diff
@@ -1,8 +1,8 @@
 #include "Message.hpp"
 
-static const size_t commands_count = 5;
-const std::string Message::commands_name[commands_count] = {"JOIN", "NICK", "PASS", "QUIT", "USER"};
-int (*const Message::commands[commands_count])(Message&, Dependencies&) = {command_join, command_nick, command_pass, command_quit, command_user};
+static const size_t commands_count = 6;
+const std::string Message::commands_name[commands_count] = {"JOIN", "NICK", "PASS", "QUIT", "USER", "PART"};
+int (*const Message::commands[commands_count])(Message&, Dependencies&) = {command_join, command_nick, command_pass, command_quit, command_user, command_part};
```
Comment (ppiques) [inline src/Message.cpp:5]: not in alphabetical order

**Review by ppiques — CHANGES_REQUESTED** [review]
small fixes needed but otherwise lgtm

File: src/commands/part.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+	ChannelManager &channel_manager = deps.channels;
+
+	while (channel_it != channels.end())
+	{
+		_base_channel base_channel = Channel::parse(*channel_it);
+		if (channel_manager.has(base_channel.getName()) == false)
+		{
+			message.peer.sendMessage(ERR_NOSUCHCHANNEL(*channel_it));
+			throw ERR_NOSUCHCHANNEL(*channel_it);
```
Comment (vfurmane) [inline src/commands/part.cpp:23]: Should not throw

Comment (vfurmane) [reply src/commands/part.cpp:23]: Test that

File: src/commands/part.cpp [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+		}
+		else 
+		{
+			Channel &channel = deps.channels[base_channel.getName()];
+			if (channel.users.has(message.peer.getUsername()) == true)
```
Comment (vfurmane) [inline src/commands/part.cpp:28]:
```suggestion
			if (channel.users.has(message.peer.getUsername()))
```

File: src/commands/part.cpp [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+				if (message.arguments[1].empty())
+				{
+					Message reply(message.peer, message.peer.getNickname());
+					channel.sendMessage(reply);
+				}
+				else
+				{
+					Message reply(message.peer, message.arguments[1]);
+					channel.sendMessage(reply);
+				}
```
Comment (vfurmane) [inline src/commands/part.cpp:40]: Don't send the username, RFC is bad

File: src/commands/part.cpp [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+			}
+			else
+			{
+				message.peer.sendMessage(ERR_NOTONCHANNEL(*channel_it));
+				throw ERR_NOTONCHANNEL(*channel_it);
```
Comment (vfurmane) [inline src/commands/part.cpp:45]: Should not throw

**Review by vfurmane — CHANGES_REQUESTED** [review]

**Review by ppiques — APPROVED** [review]

**Pull request merged**